asyncscope.md

```diff
@@ -819,6 +819,166 @@ _See_ [Why does `async_scope` terminate in the destructor instead of blocking li
 > NOTE: async RAII could be used to signal early completion because it would be composed with other async operation lifetimes.
 > The operation being stopped would complete before the async RAII operation completed -- without any blocking.
 
+Alternative design
+==================
+
+This section describes an alternative design that would break down `async_scope` into smaller parts, providing better composability and finer-grained control.
+
+## Definitions
+```cpp
+struct async_scope {
+    async_scope();
+    ~async_scope();
+    async_scope(const async_scope&) = delete;
+    async_scope(async_scope&&) = delete;
+    async_scope& operator=(const async_scope&) = delete;
+    async_scope& operator=(async_scope&&) = delete;
+
+    template <sender S>
+    @_nest-sender_@<S> nest(S&& snd);
+
+    template <sender S>
+    [[nodiscard]]
+    @_on-empty-sender_@<S> when_empty(S&& snd) const noexcept;
+};
+```
+
+The meaning of `nest()` and `when_empty()` is kept the same.
+All other methods that are absent in this alternative design can be built on top of the remaining functionality.
+
+### `spawn()`
+
+If `scope` is an object of type `async_scope`, then `scope.spawn(snd)` is equivalent with:
+```c++
+start_detached(scope.nest(snd));
```

Comment:
Can't say I'm a fan of this. For this particular case, those things may well be equivalent, but anywhere that I see a start_detached, alarm bells will ring. Whereas when I see a spawn on an async scope, alarm bells won't ring, that's the bread-and-butter operation and I know that it's likely going to be correct.

Reply:
Yes, I think this bothers me as well. Suggesting that we use start_detached to achieve this implies that start_detached is itself a good algorithm. Yet if it is used for anything that doesn't hand ownership anywhere (like chaining *anything* after that nest call) it probably isn't. So we end up in a situation where we recommend to not use start_detached except where its input is a nest-sender. In which case it's an async_scope algorithm and it might as well be explicitly tied to the async_scope.

Reply:
It makes a lot of sense. I would also want us to ensure that we don't use `start_detached` without some clear lifetime safety nets.

I see two good ways out of this:
1) completely remove `start_detached`
2) make `start_detached` work by default only with `async_scope` senders; i.e., disallow general senders, unless somebody makes an explicit specialization

The alternative of keeping `start_detached` and saying that it should never be used, seems less ideal.

I feel this is the main question we need to answer before choosing how `asnyc_scope` should be used.

Do you think it's worth writing a quick paper on this, to get a general direction from LEWG? Or should we just put this question in this paper?